kdbRUBY_wrap causing errors on build

Trying to package elektra v0.9.9 (and ongoing process lol) https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/30948 fails since a couple of days ont he ruby tests (example build log: https://gitlab.alpinelinux.org/oliver/aports/-/jobs/746078)

## Steps to Reproduce the Problem
I use https://gitlab.com/esbs/package_builder-alpine/ to create a dockerized build environment ...
```sh
package_builder-alpine.sh
```

## Expected Result
Successful compilation

## Actual Result
```
[ 37%] Building C object src/bindings/glib/CMakeFiles/gelektra-5.0.dir/gelektra-keyset.c.o
In file included from /usr/include/ruby-3.1.0/ruby.h:38,
                 from /workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx:880:
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx: In function 'VALUE _wrap_new_Key__SWIG_2(int, VALUE*, VALUE)':
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx:5405:25: error: cannot convert 'std::nullptr_t' to 'const VALUE' {aka 'const long unsigned int'} in initialization
 5405 |     hash_size = NUM2INT(rb_funcall(argv[1], rb_intern("size"), 0, NULL));
      |                         ^~~~~~~~~~
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx:5406:16: error: cannot convert 'std::nullptr_t' to 'const VALUE' {aka 'const long unsigned int'} in initialization
 5406 |     keys_arr = rb_funcall(argv[1], rb_intern("keys"), 0, NULL);
      |                ^~~~~~~~~~
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx:5414:15: error: cannot convert 'std::nullptr_t' to 'const VALUE' {aka 'const long unsigned int'} in initialization
 5414 |         key = rb_funcall(key, rb_intern("to_s"), 0, NULL);
      |               ^~~~~~~~~~
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx:5442:15: error: cannot convert 'std::nullptr_t' to 'const VALUE' {aka 'const long unsigned int'} in initialization
 5442 |         key = rb_funcall(key, rb_intern("to_s"), 0, NULL);
      |               ^~~~~~~~~~
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx:5443:15: error: cannot convert 'std::nullptr_t' to 'const VALUE' {aka 'const long unsigned int'} in initialization
 5443 |         val = rb_funcall(val, rb_intern("to_s"), 0, NULL);
      |               ^~~~~~~~~~
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx: In function 'VALUE _wrap_Key_has_metaq___(int, VALUE*, VALUE)':
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx:6468:15: error: cannot convert 'std::nullptr_t' to 'const VALUE' {aka 'const long unsigned int'} in initialization
 6468 |     argv[0] = rb_funcall(argv[0], rb_intern("to_s"), 0, NULL);
      |               ^~~~~~~~~~
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx: In function 'VALUE _wrap_Key_del_meta(int, VALUE*, VALUE)':
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx:6513:15: error: cannot convert 'std::nullptr_t' to 'const VALUE' {aka 'const long unsigned int'} in initialization
 6513 |     argv[0] = rb_funcall(argv[0], rb_intern("to_s"), 0, NULL);
      |               ^~~~~~~~~~
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx: In function 'VALUE _wrap_Key_copy_meta(int, VALUE*, VALUE)':
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx:6568:15: error: cannot convert 'std::nullptr_t' to 'const VALUE' {aka 'const long unsigned int'} in initialization
 6568 |     argv[1] = rb_funcall(argv[1], rb_intern("to_s"), 0, NULL);
      |               ^~~~~~~~~~
[ 37%] Linking C executable ../../bin/testmod_network
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx: In function 'VALUE _wrap_Key_set_meta(int, VALUE*, VALUE)':
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx:7578:15: error: cannot convert 'std::nullptr_t' to 'const VALUE' {aka 'const long unsigned int'} in initialization
 7578 |     argv[0] = rb_funcall(argv[0], rb_intern("to_s"), 0, NULL);
      |               ^~~~~~~~~~
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx: In function 'VALUE _wrap_Key_get_meta(int, VALUE*, VALUE)':
/workdir/src/libelektra-0.9.9/build/src/bindings/swig/ruby/kdbRUBY_wrap.cxx:7633:15: error: cannot convert 'std::nullptr_t' to 'const VALUE' {aka 'const long unsigned int'} in initialization
 7633 |     argv[0] = rb_funcall(argv[0], rb_intern("to_s"), 0, NULL);
```

## System Information
- Elektra: v0.9.9
- Operating System: Alpine Linux

```
(1/290) Installing java-common (0.5-r0)
(2/290) Installing libffi (3.4.2-r1)
(3/290) Installing p11-kit (0.24.1-r0)
(4/290) Installing libtasn1 (4.18.0-r0)
(5/290) Installing p11-kit-trust (0.24.1-r0)
(6/290) Installing java-cacerts (1.0-r1)
(7/290) Installing openjdk17-jre-headless (17.0.3_p7-r2)
(8/290) Installing libxau (1.0.9-r0)
(9/290) Installing libxdmcp (1.1.3-r0)
(10/290) Installing libxcb (1.15-r0)
(11/290) Installing libx11 (1.8-r0)
(12/290) Installing libxext (1.3.4-r0)
(13/290) Installing libxi (1.8-r0)
(14/290) Installing libxrender (0.9.10-r3)
(15/290) Installing libxtst (1.2.3-r3)
(16/290) Installing alsa-lib (1.2.6.1-r0)
(17/290) Installing libbz2 (1.0.8-r1)
(18/290) Installing libpng (1.6.37-r1)
(19/290) Installing freetype (2.12.1-r0)
(20/290) Installing giflib (5.2.1-r0)
(21/290) Installing libjpeg-turbo (2.1.3-r1)
(22/290) Installing lcms2 (2.13.1-r0)
(23/290) Installing openjdk17-jre (17.0.3_p7-r2)
(24/290) Installing openjdk17-jmods (17.0.3_p7-r2)
(25/290) Installing openjdk17-jdk (17.0.3_p7-r2)
(26/290) Installing gradle (7.4.2-r0)
(27/290) Installing rust-stdlib (1.60.0-r2)
(28/290) Installing xz-libs (5.2.5-r1)
(29/290) Installing libxml2 (2.9.14-r0)
(30/290) Installing llvm13-libs (13.0.1-r2)
(31/290) Installing rust (1.60.0-r2)
(32/290) Installing cargo (1.60.0-r2)
(33/290) Installing augeas-libs (1.12.0-r0)
(34/290) Installing zlib-dev (1.2.12-r1)
(35/290) Installing xz-dev (5.2.5-r1)
(36/290) Installing libxml2-dev (2.9.14-r0)
(37/290) Installing augeas-dev (1.12.0-r0)
(38/290) Installing boost1.78-atomic (1.78.0-r1)
(39/290) Installing boost1.78-chrono (1.78.0-r1)
(40/290) Installing boost1.78-container (1.78.0-r1)
(41/290) Installing boost1.78-context (1.78.0-r1)
(42/290) Installing boost1.78-contract (1.78.0-r1)
(43/290) Installing boost1.78-coroutine (1.78.0-r1)
(44/290) Installing boost1.78-date_time (1.78.0-r1)
(45/290) Installing boost1.78-fiber (1.78.0-r1)
(46/290) Installing boost1.78-filesystem (1.78.0-r1)
(47/290) Installing boost1.78-graph (1.78.0-r1)
(48/290) Installing boost1.78-iostreams (1.78.0-r1)
(49/290) Installing boost1.78-locale (1.78.0-r1)
(50/290) Installing boost1.78-thread (1.78.0-r1)
(51/290) Installing boost1.78-log (1.78.0-r1)
(52/290) Installing boost1.78-log_setup (1.78.0-r1)
(53/290) Installing boost1.78-math (1.78.0-r1)
(54/290) Installing boost1.78-prg_exec_monitor (1.78.0-r1)
(55/290) Installing boost1.78-program_options (1.78.0-r1)
(56/290) Installing gdbm (1.23-r0)
(57/290) Installing mpdecimal (2.5.1-r1)
(58/290) Installing ncurses-terminfo-base (6.3_p20220521-r0)
(59/290) Installing ncurses-libs (6.3_p20220521-r0)
(60/290) Installing readline (8.1.2-r0)
(61/290) Installing sqlite-libs (3.38.5-r0)
(62/290) Installing python3 (3.10.4-r0)
(63/290) Installing boost1.78-python3 (1.78.0-r1)
(64/290) Installing boost1.78-random (1.78.0-r1)
(65/290) Installing boost1.78-regex (1.78.0-r1)
(66/290) Installing boost1.78-serialization (1.78.0-r1)
(67/290) Installing boost1.78-stacktrace_basic (1.78.0-r1)
(68/290) Installing boost1.78-stacktrace_noop (1.78.0-r1)
(69/290) Installing boost1.78-system (1.78.0-r1)
(70/290) Installing boost1.78-timer (1.78.0-r1)
(71/290) Installing boost1.78-type_erasure (1.78.0-r1)
(72/290) Installing boost1.78-unit_test_framework (1.78.0-r1)
(73/290) Installing boost1.78-wave (1.78.0-r1)
(74/290) Installing boost1.78-wserialization (1.78.0-r1)
(75/290) Installing boost1.78-json (1.78.0-r1)
(76/290) Installing boost1.78-nowide (1.78.0-r1)
(77/290) Installing boost1.78-libs (1.78.0-r1)
(78/290) Installing boost1.78 (1.78.0-r1)
(79/290) Installing linux-headers (5.16.7-r1)
(80/290) Installing bzip2-dev (1.0.8-r1)
(81/290) Installing icu-data-full (71.1-r2)
(82/290) Installing icu-libs (71.1-r2)
(83/290) Installing icu (71.1-r2)
(84/290) Installing icu-dev (71.1-r2)
(85/290) Installing boost1.78-dev (1.78.0-r1)
(86/290) Installing boost-dev (1.78.0-r1)
(87/290) Installing clang-libs (13.0.1-r1)
(88/290) Installing clang (13.0.1-r1)
(89/290) Installing clang-extra-tools (13.0.1-r1)
(90/290) Installing clang-dev (13.0.1-r1)
(91/290) Installing lz4-libs (1.9.3-r1)
(92/290) Installing zstd-libs (1.5.2-r1)
(93/290) Installing libarchive (3.6.1-r0)
(94/290) Installing rhash-libs (1.4.2-r2)
(95/290) Installing libuv (1.44.1-r0)
(96/290) Installing cmake (3.23.1-r0)
(97/290) Installing openssl-dev (1.1.1o-r0)
(98/290) Installing nghttp2-dev (1.47.0-r0)
(99/290) Installing brotli-dev (1.0.9-r6)
(100/290) Installing curl-dev (7.83.1-r1)
(101/290) Installing libblkid (2.38-r1)
(102/290) Installing libuuid (2.38-r1)
(103/290) Installing libfdisk (2.38-r1)
(104/290) Installing libmount (2.38-r1)
(105/290) Installing libsmartcols (2.38-r1)
(106/290) Installing util-linux-dev (2.38-r1)
(107/290) Installing dbus-libs (1.14.0-r1)
(108/290) Installing dbus-dev (1.14.0-r1)
(109/290) Installing discount-libs (2.2.7b-r0)
(110/290) Installing discount (2.2.7b-r0)
(111/290) Installing discount-dev (2.2.7b-r0)
(112/290) Installing fts (1.2.7-r1)
(113/290) Installing fts-dev (1.2.7-r1)
(114/290) Installing libpciaccess (0.16-r0)
(115/290) Installing libdrm (2.4.110-r0)
(116/290) Installing libpciaccess-dev (0.16-r0)
(117/290) Installing libdrm-dev (2.4.110-r0)
(118/290) Installing libgpg-error (1.45-r0)
(119/290) Installing libgcrypt (1.10.1-r0)
(120/290) Installing libgpg-error-dev (1.45-r0)
(121/290) Installing libgcrypt-dev (1.10.1-r0)
(122/290) Installing libssh2 (1.10.0-r2)
(123/290) Installing libssh2-dev (1.10.0-r2)
(124/290) Installing http-parser (2.9.4-r0)
(125/290) Installing http-parser-dev (2.9.4-r0)
(126/290) Installing pcre (8.45-r2)
(127/290) Installing libgit2 (1.4.3-r0)
(128/290) Installing libpcre16 (8.45-r2)
(129/290) Installing libpcre32 (8.45-r2)
(130/290) Installing libpcrecpp (8.45-r2)
(131/290) Installing pcre-dev (8.45-r2)
(132/290) Installing libgit2-dev (1.4.3-r0)
(133/290) Installing llvm13 (13.0.1-r2)
(134/290) Installing py3-appdirs (1.4.4-r3)
(135/290) Installing py3-more-itertools (8.13.0-r0)
(136/290) Installing py3-ordered-set (4.0.2-r3)
(137/290) Installing py3-parsing (2.4.7-r3)
(138/290) Installing py3-six (1.16.0-r1)
(139/290) Installing py3-packaging (21.3-r0)
(140/290) Installing py3-setuptools (59.4.0-r0)
(141/290) Installing llvm13-test-utils (13.0.1-r2)
(142/290) Installing llvm13-dev (13.0.1-r2)
(143/290) Installing linenoise (1.0-r1)
(144/290) Installing lua5.3-libs (5.3.6-r3)
(145/290) Installing lua5.3 (5.3.6-r3)
(146/290) Installing lua5.3-dev (5.3.6-r3)
(147/290) Installing gettext-asprintf (0.21-r2)
(148/290) Installing libintl (0.21-r2)
(149/290) Installing libunistring (1.0-r0)
(150/290) Installing gettext-libs (0.21-r2)
(151/290) Installing gettext (0.21-r2)
(152/290) Installing gettext-dev (0.21-r2)
(153/290) Installing libxml2-utils (2.9.14-r0)
(154/290) Installing libxslt (1.1.35-r0)
(155/290) Installing docbook-xml (4.5-r6)
Executing docbook-xml-4.5-r6.post-install
(156/290) Installing docbook-xsl (1.79.2-r4)
Executing docbook-xsl-1.79.2-r4.post-install
(157/290) Installing glib (2.72.1-r0)
(158/290) Installing libffi-dev (3.4.2-r1)
(159/290) Installing glib-dev (2.72.1-r0)
(160/290) Installing python3-dev (3.10.4-r0)
(161/290) Installing libice (1.0.10-r0)
(162/290) Installing libsm (1.2.3-r0)
(163/290) Installing libxt (1.2.1-r0)
(164/290) Installing libxmu (1.1.3-r0)
(165/290) Installing xset (1.2.4-r0)
(166/290) Installing xprop (1.2.5-r0)
(167/290) Installing xdg-utils (1.1.3-r3)
(168/290) Installing libpcre2-16 (10.40-r0)
(169/290) Installing qt5-qtbase (5.15.4_git20220511-r1)
(170/290) Installing qt5-qtbase-sqlite (5.15.4_git20220511-r1)
(171/290) Installing unixodbc (2.3.11-r0)
(172/290) Installing qt5-qtbase-odbc (5.15.4_git20220511-r1)
(173/290) Installing libpq (14.3-r0)
(174/290) Installing qt5-qtbase-postgresql (5.15.4_git20220511-r1)
(175/290) Installing mariadb-connector-c (3.1.13-r4)
(176/290) Installing qt5-qtbase-mysql (5.15.4_git20220511-r1)
(177/290) Installing freetds (1.3.10-r0)
(178/290) Installing qt5-qtbase-tds (5.15.4_git20220511-r1)
(179/290) Installing hicolor-icon-theme (0.17-r1)
(180/290) Installing mesa (21.3.8-r1)
(181/290) Installing wayland-libs-server (1.20.0-r0)
(182/290) Installing mesa-gbm (21.3.8-r1)
(183/290) Installing mesa-glapi (21.3.8-r1)
(184/290) Installing wayland-libs-client (1.20.0-r0)
(185/290) Installing libxfixes (6.0.0-r0)
(186/290) Installing libxxf86vm (1.1.4-r2)
(187/290) Installing libxshmfence (1.3-r1)
(188/290) Installing mesa-gl (21.3.8-r1)
(189/290) Installing qt5-qtdeclarative (5.15.4_git20220514-r0)
(190/290) Installing libxcomposite (0.4.5-r0)
(191/290) Installing fontconfig (2.14.0-r0)
(192/290) Installing wayland-libs-cursor (1.20.0-r0)
(193/290) Installing wayland-libs-egl (1.20.0-r0)
(194/290) Installing xkeyboard-config (2.35.1-r0)
(195/290) Installing libxkbcommon (1.4.1-r0)
(196/290) Installing qt5-qtwayland (5.15.4_git20220511-r0)
(197/290) Installing mesa-egl (21.3.8-r1)
(198/290) Installing avahi-libs (0.8-r6)
(199/290) Installing nettle (3.7.3-r0)
(200/290) Installing gnutls (3.7.6-r0)
(201/290) Installing cups-libs (2.4.2-r0)
(202/290) Installing graphite2 (1.3.14-r1)
(203/290) Installing harfbuzz (4.3.0-r0)
(204/290) Installing libevdev (1.12.1-r0)
(205/290) Installing mtdev (1.1.6-r0)
(206/290) Installing eudev-libs (3.2.11-r0)
(207/290) Installing libinput-libs (1.20.1-r0)
(208/290) Installing xcb-util-wm (0.4.1-r1)
(209/290) Installing xcb-util (0.4.0-r3)
(210/290) Installing xcb-util-image (0.4.0-r1)
(211/290) Installing xcb-util-keysyms (0.4.0-r1)
(212/290) Installing xcb-util-renderutil (0.3.9-r1)
(213/290) Installing libxkbcommon-x11 (1.4.1-r0)
(214/290) Installing qt5-qtbase-x11 (5.15.4_git20220511-r1)
(215/290) Installing expat-dev (2.4.8-r0)
(216/290) Installing libpng-dev (1.6.37-r1)
(217/290) Installing freetype-dev (2.12.1-r0)
(218/290) Installing fontconfig-dev (2.14.0-r0)
(219/290) Installing xorgproto (2022.1-r0)
(220/290) Installing libice-dev (1.0.10-r0)
(221/290) Installing libsm-dev (1.2.3-r0)
(222/290) Installing libxau-dev (1.0.9-r0)
(223/290) Installing xcb-proto (1.15-r0)
(224/290) Installing libxdmcp-dev (1.1.3-r0)
(225/290) Installing libxcb-dev (1.15-r0)
(226/290) Installing xtrans (1.4.0-r1)
(227/290) Installing libx11-dev (1.8-r0)
(228/290) Installing libxext-dev (1.3.4-r0)
(229/290) Installing libxdamage (1.1.5-r1)
(230/290) Installing libxfixes-dev (6.0.0-r0)
(231/290) Installing libxdamage-dev (1.1.5-r1)
(232/290) Installing libxshmfence-dev (1.3-r1)
(233/290) Installing mesa-gles (21.3.8-r1)
(234/290) Installing mesa-osmesa (21.3.8-r1)
(235/290) Installing mesa-xatracker (21.3.8-r1)
(236/290) Installing libxxf86vm-dev (1.1.4-r2)
(237/290) Installing mesa-dev (21.3.8-r1)
(238/290) Installing perl (5.34.1-r0)
(239/290) Installing perl-error (0.17029-r1)
(240/290) Installing perl-git (2.36.1-r0)
(241/290) Installing git-perl (2.36.1-r0)
(242/290) Installing sqlite-dev (3.38.5-r0)
(243/290) Installing qt5-qtbase-dev (5.15.4_git20220511-r1)
(244/290) Installing qt5-qtdeclarative-dev (5.15.4_git20220514-r0)
(245/290) Installing qt5-qtsvg (5.15.4_git20220511-r0)
(246/290) Installing qt5-qtsvg-dev (5.15.4_git20220511-r0)
(247/290) Installing libucontext (1.2-r0)
(248/290) Installing yaml (0.2.5-r0)
(249/290) Installing ruby-libs (3.1.2-r0)
(250/290) Installing ruby (3.1.2-r0)
(251/290) Installing ruby-rdoc (3.1.2-r0)
(252/290) Installing libgmpxx (6.2.1-r2)
(253/290) Installing gmp-dev (6.2.1-r2)
(254/290) Installing libucontext-dev (1.2-r0)
(255/290) Installing ruby-dev (3.1.2-r0)
(256/290) Installing gc (8.2.0-r0)
(257/290) Installing guile-libs (3.0.8-r0)
(258/290) Installing guile (3.0.8-r0)
(259/290) Installing swig (4.0.2-r5)
(260/290) Installing xerces-c (3.2.3-r3)
(261/290) Installing xerces-c-dev (3.2.3-r3)
(262/290) Installing yajl (2.1.0-r4)
(263/290) Installing yajl-dev (2.1.0-r4)
(264/290) Installing yaml-cpp (0.7.0-r0)
(265/290) Installing yaml-cpp-dev (0.7.0-r0)
(266/290) Installing libsodium (1.0.18-r0)
(267/290) Installing libzmq (4.3.4-r0)
(268/290) Installing libsodium-dev (1.0.18-r0)
(269/290) Installing zeromq-dev (4.3.4-r0)
(270/290) Installing bash (5.1.16-r2)
Executing bash-5.1.16-r2.post-install
(271/290) Installing libassuan (2.5.5-r0)
(272/290) Installing libcap (2.64-r0)
(273/290) Installing pinentry (1.2.0-r0)
Executing pinentry-1.2.0-r0.post-install
(274/290) Installing gnupg-gpgconf (2.2.35-r3)
(275/290) Installing gpg (2.2.35-r3)
(276/290) Installing npth (1.6-r1)
(277/290) Installing gpg-agent (2.2.35-r3)
(278/290) Installing gpg-wks-server (2.2.35-r3)
(279/290) Installing libksba (1.6.0-r0)
(280/290) Installing gpgsm (2.2.35-r3)
(281/290) Installing gpgv (2.2.35-r3)
(282/290) Installing libsasl (2.1.28-r0)
(283/290) Installing libldap (2.6.2-r0)
(284/290) Installing gnupg-dirmngr (2.2.35-r3)
(285/290) Installing gnupg-utils (2.2.35-r3)
(286/290) Installing gnupg-wks-client (2.2.35-r3)
(287/290) Installing gnupg (2.2.35-r3)
(288/290) Installing ruby-power_assert (2.0.1-r0)
(289/290) Installing ruby-test-unit (3.5.3-r0)
```

## Further Log Files and Output
```
>>> elektra: Unpacking /var/cache/distfiles/libelektra-0.9.9.tar.gz...
-- The C compiler identification is GNU 11.2.1
-- The CXX compiler identification is GNU 11.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test HAS_CXX_STD
-- Performing Test HAS_CXX_STD - Success
-- compiler/linker accepts version script? TRUE
-- compiler/linker supports symbol versioning? TRUE
-- GCC detected
-- Performing Test HAS_CFLAG_MAYBE_UNINITIALIZED
-- Performing Test HAS_CFLAG_MAYBE_UNINITIALIZED - Success
-- C flags are -Os -fomit-frame-pointer -std=gnu99  -Wno-deprecated-declarations  -Wstrict-prototypes  -Wno-long-long -Wpedantic -Wno-variadic-macros -Wall -Wextra -Wno-overlength-strings -Wsign-compare -Wfloat-equal -Wformat -Wformat-security -Wshadow -Wcomments -Wtrigraphs -Wundef -Wuninitialized -Winit-self -Wmaybe-uninitialized -Wsign-compare -Wfloat-equal
-- CXX flags are -Os -fomit-frame-pointer -std=c++11  -Wno-deprecated-declarations  -Wold-style-cast -Wstrict-null-sentinel -D_GLIBCXX_USE_NANOSLEEP -Wno-missing-field-initializers -Woverloaded-virtual  -Wsign-promo  -Wno-long-long -Wpedantic -Wno-variadic-macros -Wall -Wextra -Wno-overlength-strings -Wsign-compare -Wfloat-equal -Wformat -Wformat-security -Wshadow -Wcomments -Wtrigraphs -Wundef -Wuninitialized -Winit-self -Wmaybe-uninitialized
-- You are building Elektra 0.9.9
-- Looking for __GNU_LIBRARY__
-- Looking for __GNU_LIBRARY__ - not found
-- Exclude plugin simpleini because simpleini needs glibc to work, use the mini plugin instead
-- Include plugin augeas
-- Include plugin base64
-- Include plugin blacklist
-- Include plugin c
-- Include plugin ccode
-- Include plugin conditionals
-- Include plugin constants
-- Include plugin counter
-- Include plugin crypto
-- Include plugin csvstorage
-- Include plugin curlget
-- Include plugin date
-- Include plugin dbus
-- Include plugin desktop
-- Include plugin directoryvalue
-- Include plugin dpkg
-- Include plugin dump
-- Include plugin error
-- Include plugin fcrypt
-- Performing Test Iconv_IS_BUILT_IN
-- Performing Test Iconv_IS_BUILT_IN - Success
-- Include plugin filecheck
-- Looking for setmntent
-- Looking for setmntent - found
-- Include plugin fstab
-- Include plugin gitresolver
-- Include plugin glob
-- gopts: using gopts_procfs.h
-- Include plugin gopts
-- Include plugin hexcode
-- Include plugin hexnumber
-- Include plugin hosts
-- Include plugin iconv
-- Include plugin ipaddr
-- Found JNI: /usr/lib/jvm/default-jvm/jre/lib/libjawt.so  
-- Found Threads: TRUE  
-- Looking for JNI_VERSION_10
-- Looking for JNI_VERSION_10 - found
-- Include plugin jni
-- Include plugin keytometa
-- Include plugin line
-- Include plugin lineendings
-- Include plugin list
-- Include plugin logchange
CMake Deprecation Warning at scripts/cmake/Modules/LibAddMacros.cmake:197 (cmake_policy):
  The OLD behavior for policy CMP0078 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
Call Stack (most recent call first):
  src/plugins/lua/CMakeLists.txt:6 (find_swig)


-- Include plugin lua
-- Include plugin macaddr
-- Include plugin mathcheck
-- Include plugin mini
-- Include plugin mmapstorage
-- Include plugin multifile
-- Include plugin network
-- Include plugin ni
-- Include plugin noresolver
-- Looking for euidaccess
-- Looking for euidaccess - found
-- Looking for access
-- Looking for access - found
-- Looking for geteuid
-- Looking for geteuid - found
-- Looking for getegid
-- Looking for getegid - found
-- Looking for seteuid
-- Looking for seteuid - found
-- Looking for setegid
-- Looking for setegid - found
-- Looking for getpwnam
-- Looking for getpwnam - found
-- Looking for getpwuid
-- Looking for getpwuid - found
-- Looking for getgrouplist
-- Looking for getgrouplist - found
-- Looking for getgrgid
-- Looking for getgrgid - found
-- Include plugin path
-- Include plugin profile
-- Looking for mkfifo
-- Looking for mkfifo - found
-- Looking for fork
-- Looking for fork - found
CMake Deprecation Warning at scripts/cmake/Modules/LibAddMacros.cmake:197 (cmake_policy):
  The OLD behavior for policy CMP0078 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
Call Stack (most recent call first):
  src/plugins/python/CMakeLists.txt:7 (find_swig)


<string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
-- Include plugin python
-- Include plugin quickdump
-- Include plugin range
-- Include plugin reference
-- Include plugin rename
-- Include plugin resolver_fm_hpu_b
-- Include plugin resolver_fm_b_b
-- Include plugin resolver_fm_pb_b
-- Include plugin resolver_fm_hb_b
-- Include plugin resolver_fm_hp_b
-- Include plugin resolver_fm_ub_x
-- Include plugin resolver_fm_xb_x
-- Include plugin resolver_fm_xp_x
-- Include plugin resolver_fm_xhp_x
-- Include plugin resolver_fm_uhb_xb
-- Include plugin rgbcolor
CMake Deprecation Warning at scripts/cmake/Modules/LibAddMacros.cmake:197 (cmake_policy):
  The OLD behavior for policy CMP0078 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
Call Stack (most recent call first):
  src/plugins/ruby/CMakeLists.txt:16 (find_swig)


-- Include plugin ruby
-- Include plugin shell
-- Include plugin spec
-- Include plugin specload
-- Include plugin sync
-- Include plugin syslog
-- Include plugin timeofday
-- Include plugin tracer
-- Include plugin type
-- Include plugin uname
-- Include plugin unit
-- Include plugin validation
-- Include plugin xerces
-- Include plugin xmltool
-- Include plugin yajl
-- Include plugin yamlcpp
-- Looking for fnmatch
-- Looking for fnmatch - found
-- Exclude Library notification because internalnotification plugin excluded
-- Include binding cpp
-- Include binding jna
-- Include binding rust for shared builds
CMake Deprecation Warning at scripts/cmake/Modules/LibAddMacros.cmake:197 (cmake_policy):
  The OLD behavior for policy CMP0078 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
Call Stack (most recent call first):
  src/bindings/swig/CMakeLists.txt:18 (find_swig)


CMake Deprecation Warning at src/bindings/swig/python/CMakeLists.txt:8 (cmake_policy):
  The OLD behavior for policy CMP0078 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- Include binding python
<string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
-- Include binding lua
CMake Warning (dev) at /usr/share/cmake/Modules/UseSWIG.cmake:775 (message):
  Policy CMP0078 is not set: UseSWIG generates standard target names.  Run
  "cmake --help-policy CMP0078" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

Call Stack (most recent call first):
  src/bindings/swig/lua/CMakeLists.txt:30 (swig_add_library)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /usr/share/cmake/Modules/UseSWIG.cmake:617 (message):
  Policy CMP0086 is not set: UseSWIG honors SWIG_MODULE_NAME via -module
  flag.  Run "cmake --help-policy CMP0086" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  /usr/share/cmake/Modules/UseSWIG.cmake:888 (SWIG_ADD_SOURCE_TO_MODULE)
  src/bindings/swig/lua/CMakeLists.txt:30 (swig_add_library)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Include binding ruby
CMake Warning (dev) at /usr/share/cmake/Modules/UseSWIG.cmake:775 (message):
  Policy CMP0078 is not set: UseSWIG generates standard target names.  Run
  "cmake --help-policy CMP0078" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

Call Stack (most recent call first):
  src/bindings/swig/ruby/CMakeLists.txt:101 (swig_add_library)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /usr/share/cmake/Modules/UseSWIG.cmake:617 (message):
  Policy CMP0086 is not set: UseSWIG honors SWIG_MODULE_NAME via -module
  flag.  Run "cmake --help-policy CMP0086" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  /usr/share/cmake/Modules/UseSWIG.cmake:888 (SWIG_ADD_SOURCE_TO_MODULE)
  src/bindings/swig/ruby/CMakeLists.txt:101 (swig_add_library)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /usr/share/cmake/Modules/UseSWIG.cmake:775 (message):
  Policy CMP0078 is not set: UseSWIG generates standard target names.  Run
  "cmake --help-policy CMP0078" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

Call Stack (most recent call first):
  src/bindings/swig/ruby/CMakeLists.txt:114 (swig_add_library)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /usr/share/cmake/Modules/UseSWIG.cmake:617 (message):
  Policy CMP0086 is not set: UseSWIG honors SWIG_MODULE_NAME via -module
  flag.  Run "cmake --help-policy CMP0086" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  /usr/share/cmake/Modules/UseSWIG.cmake:888 (SWIG_ADD_SOURCE_TO_MODULE)
  src/bindings/swig/ruby/CMakeLists.txt:114 (swig_add_library)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Exclude binding gsettings because excluded by category EXPERIMENTAL
-- Include binding glib
-- Exclude binding intercept_fs because excluded by category EXPERIMENTAL
-- Exclude binding intercept_env because intercept_env currently needs glibc
-- Exclude binding io_uv because excluded by category EXPERIMENTAL
-- Exclude binding io_ev because excluded by category EXPERIMENTAL
-- Exclude binding io_glib because excluded by category EXPERIMENTAL
-- Include tool kdb
-- Include tool qt-gui
CMake Warning (dev) at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (DISCOUNT)
  does not match the name of the calling package (Discount).  This can lead
  to problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  scripts/cmake/Modules/FindDiscount.cmake:73 (find_package_handle_standard_args)
  src/tools/qt-gui/CMakeLists.txt:19 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Looking for clearenv
-- Looking for clearenv - found
-- Looking for setenv
-- Looking for setenv - found
-- Looking for futimens
-- Looking for futimens - found
-- Looking for hsearch_r
-- Looking for hsearch_r - found
-- Looking for futimes
-- Looking for futimes - found
-- Looking for glob
-- Looking for glob - found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for ctype.h
-- Looking for ctype.h - found
-- Looking for errno.h
-- Looking for errno.h - found
-- Looking for features.h
-- Looking for features.h - found
-- Looking for locale.h
-- Looking for locale.h - found
-- Looking for stdio.h
-- Looking for stdio.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for time.h
-- Looking for time.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of int
-- Check size of int - done
-- Check size of long
-- Check size of long - done
-- Check size of size_t
-- Check size of size_t - done
-- Check size of long long
-- Check size of long long - done
-- Check size of long double
-- Check size of long double - done
-- Check size of mode_t
-- Check size of mode_t - done
-- Check size of float
-- Check size of float - done
-- Check size of double
-- Check size of double - done
-- Check size of ((struct timeval*)0)->tv_usec
-- Check size of ((struct timeval*)0)->tv_usec - done
-- Check size of ((struct stat*)0)->st_size
-- Check size of ((struct stat*)0)->st_size - done
CMake Deprecation Warning at CMakeLists.txt:4 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


Cloning into 'googletest-src'...
HEAD is now at e2239ee6 Googletest export
-- Found Python: /usr/bin/python3.10 (found version "3.10.4") found components: Interpreter 
-- Looking for nftw
-- Looking for nftw - found
-- Looking for mkdtemp
-- Looking for mkdtemp - found
-- Exclude component libelektra5-zeromq because zeromqrecv, zeromqsend plugins excluded.
-- Exclude component libelektra5-journald because journald plugin excluded.
-- Exclude component io-ev-elektra because io_ev binding excluded.
-- Exclude component io-glib-elektra because io_glib binding excluded.
-- Exclude component io-uv-elektra because io_uv binding excluded.
-- Detected Alpine-Linux3.16.0-amd64. Use make package to build packages (TBZ2).
-- Configuring done
CMake Warning (dev) at src/libs/pluginprocess/tests/CMakeLists.txt:16 (add_executable):
  Policy CMP0115 is not set: Source file extensions must be explicit.  Run
  "cmake --help-policy CMP0115" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  File:

    /workdir/src/libelektra-0.9.9/src/libs/pluginprocess/tests/testlib_pluginprocess.c
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
```
```

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -71,11 +71,9 @@ The following section lists news about the [plugins](https://www.libelektra.org/
 - <<TODO>>
 - <<TODO>>
 
-### <<Plugin>>
+### date
 
-- <<TODO>>
-- <<TODO>>
-- <<TODO>>
+- Exclude the tests for formats that require GNU extensions of `strptime` on non-GNU systems. _(@kodebach)_
 
 ### <<Plugin>>
 
@@ -180,11 +178,9 @@ you up-to-date with the multi-language support provided by Elektra.
 - Fixed missing Javadoc in Java Sorted plugin _(Michael Tucek @tucek)_
 - <<TODO>>
 
-### <<Binding>>
+### Ruby
 
-- <<TODO>>
-- <<TODO>>
-- <<TODO>>
+- Replace `NULL` in `rb_funcall` with `Qnil` to avoid compiler errors/warnings on some systems. _(@kodebach)_
 
 ### <<Binding>>
 

--- a/src/bindings/swig/ruby/kdb.i
+++ b/src/bindings/swig/ruby/kdb.i
@@ -263,7 +263,7 @@ underlaying key, which allows a Ruby-style iteration over metadata:
  * cases this is useful, to allow passing in Symbols as meta names. */
 %typemap(in) (const std::string & metaName) {
   // typemap in for getMeta
-  $input = rb_funcall($input, rb_intern("to_s"), 0, NULL);
+  $input = rb_funcall($input, rb_intern("to_s"), 0, Qnil);
   $1 = new std::string(StringValueCStr($input));
 }
 %typemap(freearg) (const std::string & metaName) {
@@ -308,8 +308,8 @@ underlaying key, which allows a Ruby-style iteration over metadata:
   /* $input substitution does not here, so we have to reference
      input variables directly */
 
-  hash_size = NUM2INT(rb_funcall(argv[1], rb_intern("size"), 0, NULL));
-  keys_arr = rb_funcall(argv[1], rb_intern("keys"), 0, NULL);
+  hash_size = NUM2INT(rb_funcall(argv[1], rb_intern("size"), 0, Qnil));
+  keys_arr = rb_funcall(argv[1], rb_intern("keys"), 0, Qnil);
   if (hash_size > 0) {
     /* first we check if we can find the "flags" key.
        this has to be passed to the kdb::Key constructor already */
@@ -317,7 +317,7 @@ underlaying key, which allows a Ruby-style iteration over metadata:
       key = rb_ary_entry(keys_arr, i);
       val = rb_hash_aref(argv[1], key);
       /* convert key to String, in case of being a Symbol */
-      key = rb_funcall(key, rb_intern("to_s"), 0, NULL);
+      key = rb_funcall(key, rb_intern("to_s"), 0, Qnil);
       /* check for flags and extract them */
       if (strcmp("flags", StringValueCStr(key)) == 0) {
         Check_Type(val, T_FIXNUM);
@@ -345,8 +345,8 @@ underlaying key, which allows a Ruby-style iteration over metadata:
     for (i = 0; i < hash_size; i++) {
       key = rb_ary_entry(keys_arr, i);
       val = rb_hash_aref(argv[1], key);
-      key = rb_funcall(key, rb_intern("to_s"), 0, NULL);
-      val = rb_funcall(val, rb_intern("to_s"), 0, NULL);
+      key = rb_funcall(key, rb_intern("to_s"), 0, Qnil);
+      val = rb_funcall(val, rb_intern("to_s"), 0, Qnil);
       /* ignore certain keys */
       if (strcmp("flags", StringValueCStr(key)) == 0) continue;
       /* 'value' has also a special meaning */

--- a/src/plugins/date/testmod_date.c
+++ b/src/plugins/date/testmod_date.c
@@ -50,6 +50,7 @@ static void testIso (const char * date, const char * isoString, const short res)
 	PLUGIN_CLOSE ();
 }
 
+#ifdef __GNU_LIBRARY__
 static void testRfc2822 (const char * date, const short res)
 {
 	Key * parentKey = keyNew ("user:/tests/date", KEY_VALUE, "", KEY_END);
@@ -64,12 +65,16 @@ static void testRfc2822 (const char * date, const short res)
 	keyDel (parentKey);
 	PLUGIN_CLOSE ();
 }
+#endif
 
 int main (int argc, char ** argv)
 {
 	printf ("DATE     TESTS\n");
 	printf ("==================\n\n");
+
+#ifdef __GNU_LIBRARY__
 	const char * old_locale = setlocale (LC_ALL, NULL);
+#endif
 
 	init (argc, argv);
 
@@ -87,17 +92,11 @@ int main (int argc, char ** argv)
 	testIso ("2016-12-12T23:59:01Z", "datetime complete", 1);
 	testIso ("2016-12-12 23:59:01Z", "datetime complete noT", 1);
 	testIso ("2016-12-12T23:59:01Z", "datetime truncated", -1);
+	testIso ("2016-12-12T23:59:01", "datetime complete", 1);
 	testIso ("-12-12T23:59:01Z", "datetime truncated", 1);
 	testIso ("22:30+04", "utc extended", 1);
 	testIso ("22:30-04", "utc extended", 1);
 	testIso ("2016-W23", "weekdate", 1);
-#else
-	testIso ("2016-12-12T23:59:01", "datetime complete", 1);
-	testIso ("2016-12-12T23:59:01", "datetime truncated", -1);
-	testIso ("-12-12T23:59:01", "datetime truncated", 1);
-#endif
-	testIso ("2230", "timeofday extended", -1);
-	testIso ("2230", "timeofday basic", 1);
 
 	setlocale (LC_ALL, "C");
 	testRfc2822 ("Sat, 01 Mar 2016 23:59:01 +0400", 1);
@@ -106,6 +105,12 @@ int main (int argc, char ** argv)
 	testRfc2822 ("01 Mar 2016 23:59 +0400", 1);
 	testRfc2822 ("01 Mar 2016 01:00:59", -1);
 	setlocale (LC_ALL, old_locale);
+#else
+	testIso ("2016-12-12T23:59:01", "datetime truncated", -1);
+	testIso ("-12-12T23:59:01", "datetime truncated", 1);
+#endif
+	testIso ("2230", "timeofday extended", -1);
+	testIso ("2230", "timeofday basic", 1);
 
 	print_result ("testmod_date");
 

